homeassistant/components/hive/binary_sensor.py

```diff
@@ -41,24 +49,35 @@ def device_info(self):
     @property
     def device_class(self):
         """Return the class of this sensor."""
-        return DEVICETYPE_DEVICE_CLASS.get(self.node_device_type)
+        return DEVICETYPE.get(self.device["hiveType"])
 
     @property
     def name(self):
         """Return the name of the binary sensor."""
-        return self.node_name
+        return self.device["haName"]
+
+    @property
+    def available(self):
+        """Return if the device is available."""
+        if self.device["hiveType"] != "Connectivity":
+            return self.device["deviceData"]["online"]
+        return True
 
     @property
     def device_state_attributes(self):
         """Show Device Attributes."""
-        return self.attributes
+        return {
+            CONF_AVAILABLE: self.attributes[CONF_AVAILABLE],
```

Comment:
Please use `ATTR_` constants for attribute keys. We conly use `CONF_` constants for config items.

Reply:
What does the `available` attribute represent?

Reply:
Fixed in latest commit

Reply:
The value of available will be string of either Online or Offline